storage/pebble/receipts.go

```diff
@@ -39,41 +38,36 @@ func NewReceipts(store *Storage) *Receipts {
 // - receipt transaction ID => block height bytes
 // - receipt block height => list of encoded receipts (1+ per block)
 // - receipt block height => list of bloom filters (1+ per block)
-func (r *Receipts) Store(receipt *models.StorageReceipt, batch *pebble.Batch) error {
+func (r *Receipts) Store(
+	receipts []*models.StorageReceipt,
+	evmHeight uint64,
+	batch *pebble.Batch,
+) error {
 	r.mux.Lock()
 	defer r.mux.Unlock()
 
-	// we must first retrieve any already saved receipts at the provided height,
-	// and if found we must add to the list, because this method can be called multiple
-	// times when indexing a single EVM height, which can contain multiple receipts
-	blockHeight := receipt.BlockNumber.Bytes()
-	receipts, err := r.getByBlockHeight(blockHeight, batch)
-	if err != nil && !errors.Is(err, errs.ErrNotFound) { // anything but not found is failure
-		return fmt.Errorf("failed to store receipt to height, retrieve exisint receipt errror: %w", err)
-	}
+	blooms := []*gethTypes.Bloom{}
+	var height []byte
 
-	// same goes for blooms
-	blooms, err := r.getBloomsByBlockHeight(blockHeight, batch)
-	if err != nil && !errors.Is(err, errs.ErrNotFound) {
-		return fmt.Errorf("failed to store receipt to height, retrieve existing blooms error: %w", err)
-	}
+	for _, receipt := range receipts {
+		if receipt.BlockNumber.Uint64() != evmHeight {
+			return fmt.Errorf("receipt belongs to different block height: %d", receipt.BlockNumber.Uint64())
+		}
```

Comment:
this check doesn't make sense anylonger, as it should be done by whoever calls this API and it is done there